src/backend/api/meals.js

```diff
@@ -2,11 +2,96 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
+
+// GET api/meals/ query parameters
+
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    if (request.query.maxPrice) {
+      const maxPrice = parseInt(request.query.maxPrice);
+      const mealsWithMaxPrice = await knex("meals").where("price", "<", maxPrice);
+      response.json(mealsWithMaxPrice);
+    }
+    else if (request.query.tittle) {
+      const givenTittle = request.query.tittle;
+      const mealsWithTittle = await knex('meals')
+        .where("tittle", 'like', '%' + givenTittle + '%');
+      response.json(mealsWithTittle);
+
+    }
+    else if (request.query.limit) {
+      const givenLimit = parseInt(request.query.limit);
+      const mealsWithLimit = await knex('meals')
+        .limit(givenLimit);
+
+      response.json(mealsWithLimit);
+
+    }
+    else if (request.query.availableReservations) {
+      const mealsWithReservations = await knex('meals')
+        .where("max_reservations", ">", 0);
+
+      response.json(mealsWithReservations);
+    }
+    else if (request.query.createdAfter) {
+      const givenDate = new Date(request.query.createdAfter);
+      const mealsCreatedAfterDate = await knex('meals').where('created_date', '>', givenDate);
+
+      response.json(mealsCreatedAfterDate);
+    }
+    else {
+      const titles = await knex("meals").select("tittle");
+      const meals = await knex("meals");
+      response.json(meals);
+    }
+  } catch (error) {
+    throw error;
+  }
+});
+
+
+// get, post, put and delete
+
+router.post("/", async (request, response) => {
+  console.log(request.body);
+  try {
+    await knex("meals")
+      .insert(request.body)
```

Comment:
It's not a requirement here but normally you would validate the input data before sending it to the DB.